pallets/rolldown/src/lib.rs

```diff
@@ -538,50 +588,50 @@ impl<T: Config> Pallet<T> {
 		let asset_id = match T::AssetRegistryProvider::get_l1_asset_id(eth_asset.clone()) {
 			Some(id) => id,
 			None => T::AssetRegistryProvider::create_l1_asset(eth_asset)
-				.or(Err("Failed to create L1 Asset"))?,
+				.or(Err(Error::<T>::L1AssetCreationFailed))?,
 		};
 		log!(info, "Deposit processed successfully: {:?}", deposit_request_details);
 
 		// ADD tokens: mint tokens for user
 		T::Tokens::mint(
 			asset_id,
 			&account,
-			amount.try_into().map_err(|_| "u128 to Balance failed")?,
+			amount.try_into().or(Err(Error::<T>::BalanceOverflow))?,
 		)?;
 		Ok(())
 	}
 
-	fn process_withdraw(withdraw_request_details: &messages::Withdraw) -> Result<(), &'static str> {
-		// fail will occur if user has not enough balance
-		let amount: u128 = withdraw_request_details
-			.amount
-			.try_into()
-			.map_err(|_| "u128 to Balance failed")?;
-		let account: T::AccountId =
-			T::AddressConverter::convert(withdraw_request_details.depositRecipient);
-
-		let eth_asset = L1Asset::Ethereum(withdraw_request_details.tokenAddress);
-		let asset_id = T::AssetRegistryProvider::get_l1_asset_id(eth_asset.clone())
-			.ok_or("L1AssetNotFound")?;
-
-		T::Tokens::ensure_can_withdraw(
-			asset_id.into(),
-			&account,
-			amount.try_into().map_err(|_| "u128 to Balance failed")?,
-			WithdrawReasons::all(),
-			Default::default(),
-		)
-		.or(Err("NotEnoughAssets"))?;
-
-		// burn tokes for user
-		T::Tokens::burn_and_settle(
-			asset_id,
-			&account,
-			amount.try_into().map_err(|_| "u128 to Balance failed")?,
-		)?;
-
-		Ok(())
-	}
+	// fn process_withdraw(withdraw_request_details: &messages::Withdraw) -> Result<(), &'static str> {
```

Comment:
why its commented out ? please either uncomment or delete. (its in git history either way)

Reply:
comment removed